packages/arb-token-bridge-ui/src/util/TokenUtils.ts

```diff
@@ -412,6 +459,14 @@ export async function isGatewayRegistered({
   parentChainProvider: Provider
   childChainProvider: Provider
 }): Promise<boolean> {
+  // skip teleport transfers because there is no custom gateway there
+  const sourceChainId = await getChainIdFromProvider(parentChainProvider)
+  const destinationChainId = await getChainIdFromProvider(childChainProvider)
+  if (isTeleport({ sourceChainId, destinationChainId })) {
```

Comment:
parent chain should be parent chain, source chain as a name allows parent or child chains, this name is not accurate and can potentially cause a very bad day for debugging

Reply:
I have added a comment here, but we only check for gateway registration for deposits where this condition holds true.   
`// note: we only check for gateway registration for deposits, hence the assumption that source:parentChain, and destination:childChain`